docs/proposals/image-spec.md

```diff
@@ -0,0 +1,480 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/image-spec.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Abstract
+
+A proposal for adding an `ImageSpec` type to container specs, to facilitate alternative image formats and more
+deterministic image references.
+
+## Motivation
+
+Currently, all of the information about where to locate an image or how to verify the image is embedded in a single
+string in the container spec. However that opaque string is not structured and sometimes confusing for users if they
+are not familiar with how the string is being parsed.
+
+For example, `gcr.io/google_containers/nginx` refers to an image located in `gcr.io` registry, with the repository path
+as `google_containers/etcd`; While `library/nginx` refers to an image located in `docker.io` registry, with the repository path
+as `library/nginx`, which is same as `nginx` because it is an [official Docker repository](https://docs.docker.com/docker-hub/official_repos/).
+
+Besides, having only one string to represent the image precludes the possibility of adding any other image types
+such as [ACI](https://github.com/appc/spec), [Fabric8 Java containers](http://fabric8.io/gitbook/javaContainer.html),
+etc.
+
+The goal of this proposal is to add an *optional* `ImageSpec` field in the `Container` object,
+which will give users the ability to:
+
+1. Express the image types they want to use.
+2. Express how and where they want the images to be fetched from.
+
+## Design
+
+1. Add an optional field called `ImageSpec` in the `Container` object and make `Image` field optional.
+2. For GET requests, We will show both the `Image` and `ImageSpec` fields for the sake of backward compatibility.
+3. For other requests, either `Image` or `ImageSpec` should be specified but not both. These two fields are
+   mutually exclusive. However `ImageSpec` will always be populated internally if `Image` is being used.
+4. If `ImageSpec` is not empty, it MUST have one and ONLY one non-empty field to specify the details of the image,
```

Comment:
Considering the fact that not all runtimes currently support all image types. should the image type be opaque in the API?